tests/scripts/all.sh

```diff
@@ -625,6 +625,18 @@ component_check_files () {
     record_status tests/scripts/check-files.py
 }
 
+component_check_changelog () {
```

Comment:
This last commit seems quite unrelated to the others. It would probably be better for it to be in a separate PR I think. But as what matters most eventually is the commit history, fine by me if you don't want to do the split.

Reply:
Adding the changelog check was the original goal of this PR. This led into making it pass (fixing the line endings in `ChangeLog.d/*`), then to arranging that it won't fail in an incomprehensible way (modifying `check-files.py` to check the line endings in these files), and from there to fixing the other files where the improved `check-files` found something to say.